### PR TITLE
[BugFix] Fix StarMgrMetaSyncer sync

### PR DESCRIPTION
**Pull request by meegoo**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

## Documentation PRs only:

If you are submitting a PR that adds or changes English documentation and have not
included Chinese documentation, then you can check the box to request GPT to translate the
English doc to Chinese. Please ensure to uncheck the **Do not translate** box if translation is needed.
The workflow will generate a new PR with the Chinese translation after this PR is merged.

- [ ] Yes, translate English markdown files with GPT
- [x] Do not translate

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java [hunk 1]
```diff
@@ -278,6 +278,8 @@ public enum OlapTableState {
     // The flag is used to indicate whether the table shard group has changed.
     public AtomicBoolean isShardGroupChanged = new AtomicBoolean(false);
 
+    public AtomicBoolean isAutomaticBucketing = new AtomicBoolean(false);
```
Comment (starrocks-xupeng) [inline fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java:281]: add comment to show these 2 vars are used by StarMgrMetaSync

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java [hunk 2]
```diff
@@ -2049,8 +2049,13 @@ public synchronized TImmutablePartitionResult updateImmutablePartitionInternal(T
                 locker.unLockDatabase(db, LockType.READ);
             }
             if (mutablePartitions.size() <= 0) {
-                GlobalStateMgr.getCurrentState().getLocalMetastore()
-                        .addSubPartitions(db, olapTable, partition, 1, warehouseId);
+                try {
```
Comment (starrocks-xupeng) [inline fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:2052]: wrap into one function?

File: fe/fe-core/src/main/java/com/starrocks/lake/StarMgrMetaSyncer.java [hunk 3]
```diff
@@ -309,7 +309,13 @@ public boolean syncTableMetaInternal(Database db, OlapTable table, boolean force
         for (PhysicalPartition physicalPartition : physicalPartitions) {
             locker.lockDatabase(db, LockType.READ);
             try {
-                // shard group might be changed by automatic bucket or schema change
+                if (table.getState() != OlapTable.OlapTableState.NORMAL) {
```
Comment (starrocks-xupeng) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarMgrMetaSyncer.java:312]: add scheme change comment

**Review by wyb — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by meegoo** [comment]
@mergify backport branch-3.3

**Pull request merged**